### ISSUE TITLE
frame is not hidden by frame hidden option within sidebyside tcolorboxes

### ISSUE THREAD

**Issue by nignasi** [T-F-S/tcolorbox#138]
Although "frame hidden" option can be used to avoid seeing the frame in bicolor boxes when one part is white. It seems that this solution doesn't work for sidebyside tcolorboxes.

This problem can be seen in https://tex.stackexchange.com/q/597504/1952

A similar code could be:

```
\documentclass{article}
\usepackage[most]{tcolorbox}

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%% 1 %%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{tcolorbox}[
    bicolor,
    frame hidden,
    colback=blue!40,
    colbacklower=white,
    boxrule=0pt,
    sidebyside,
]%
A\tcblower
B
\end{tcolorbox}

%%%%%%%%%%%%%%%%%%%%%%%% 2 %%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{tcolorbox}[
    bicolor,
    frame hidden,
    colback=blue!40,
    colbacklower=white,
    boxrule=0pt,
]%
A\tcblower
B\end{tcolorbox}

\end{document}
```
![dibuix](https://user-images.githubusercontent.com/2562430/118611745-72f08780-b7bd-11eb-8bda-a13f3e80479a.png)

**Comment by T-F-S** [comment]
I see the problem. It is correct to use `frame hidden` to remove the small frame remnants. Actually, `frame hidden` works as intended, but unfortunately, there is a second kind of frame in the background.

Currently, the code for the interior fills the whole interior from left to right, even in case of chosen `bicolor` (or `tile`). The right handed part is painted above which leaves the frame-alike remnants.

What I need to do is to add distinctions of cases for `bicolor` and `tile` with adapted drawing routines for the interior.

I'm a little behind my plans for the next `tcolorbox` version (blame it on COVID-19 and online lectures), but I expect to have time for it in the not-so-far future.

**Comment by muzimuzhi** [comment]
It's more or less a PDF viewer problem (or feature) and is reproducible with simple tikz codes:
```tex
\documentclass{article}
\usepackage{tikz}

\begin{document}

\begin{tikzpicture}[rounded corners=3pt]
  \fill[blue!40] (0,0) rectangle (1,1);
  \fill[white]   (0,0) rectangle (1,1);
\end{tikzpicture}

\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/118688165-61989100-b838-11eb-9e7c-36b9ef52cd69.png)

[Documentation of `nicematrix`](http://mirrors.ctan.org/macros/latex/contrib/nicematrix/nicematrix.pdf), sec. 6.1 contains some discussion about this.

**Comment by T-F-S** [comment]
Yes, it is PDF viewer problem. These artifacts should not be visible at all.

Nevertheless, I classified the problem as 'bug', because `tcolorbox` tries to avoid these PDF viewer problems where possible, but I missed the described case. But, the *white line problem* betweeen two areas filled with the same color will remain.

**Comment by T-F-S** [comment]
Fixed with version 4.50 (2021/05/21)
